File: app/src/main/java/com/perflyst/twire/adapters/MentionAdapter.java
package com.perflyst.twire.adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.perflyst.twire.R;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Sebastian Rask on 29-01-2017.
 */

public class MentionAdapter extends RecyclerView.Adapter<MentionAdapter.SuggestionViewHolder> {
    private final MentionAdapterDelegate mDelegate;
    private List<String> mentionSuggestions;

    public MentionAdapter(MentionAdapterDelegate aDelegate) {
        mDelegate = aDelegate;
        mentionSuggestions = new ArrayList<>();
    }

    @Override
    @NonNull
    public SuggestionViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View itemView = LayoutInflater
                .from(parent.getContext())
                .inflate(R.layout.mention_suggestion, parent, false);

        return new SuggestionViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(final SuggestionViewHolder holder, int position) {
        holder.mName.setText(mentionSuggestions.get(position));
        holder.itemView.setOnClickListener(view -> mDelegate.onSuggestionClick(holder.mName.getText().toString()));
    }

    @Override
    public int getItemCount() {
        return mentionSuggestions.size();
    }

    public void setSuggestions(List<String> suggestions) {
        mentionSuggestions = suggestions;
        notifyDataSetChanged();
    }

    public interface MentionAdapterDelegate {
        void onSuggestionClick(String suggestion);
    }

    static class SuggestionViewHolder extends RecyclerView.ViewHolder {
        final TextView mName;

        SuggestionViewHolder(View itemView) {
            super(itemView);
            mName = itemView.findViewById(R.id.txtSuggestion);
        }
    }
}

File: app/src/main/java/com/perflyst/twire/adapters/MentionAdapter.kt
package com.perflyst.twire.adapters

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.perflyst.twire.adapters.MentionAdapter.SuggestionViewHolder
import com.perflyst.twire.databinding.MentionSuggestionBinding

/**
 * Created by Sebastian Rask on 29-01-2017.
 */
class MentionAdapter(private val mDelegate: MentionAdapterDelegate) :
    RecyclerView.Adapter<SuggestionViewHolder?>() {
    private var mentionSuggestions: MutableList<String>

    init {
        mentionSuggestions = ArrayList<String>()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SuggestionViewHolder {
        val binding = MentionSuggestionBinding.inflate(
            LayoutInflater.from(parent.context), parent, false
        )
        return SuggestionViewHolder(binding)
    }

    override fun onBindViewHolder(holder: SuggestionViewHolder, position: Int) {
        holder.binding.txtSuggestion.text = mentionSuggestions[position]
        holder.itemView.setOnClickListener {
            mDelegate.onSuggestionClick(holder.binding.txtSuggestion.getText().toString())
        }
    }

    override fun getItemCount(): Int {
        return mentionSuggestions.size
    }

    @SuppressLint("NotifyDataSetChanged")
    fun setSuggestions(suggestions: MutableList<String>) {
        mentionSuggestions = suggestions
        notifyDataSetChanged()
    }

    fun interface MentionAdapterDelegate {
        fun onSuggestionClick(suggestion: String?)
    }

    class SuggestionViewHolder(binding: MentionSuggestionBinding) :
        BindingViewHolder<MentionSuggestionBinding>(binding)
}
